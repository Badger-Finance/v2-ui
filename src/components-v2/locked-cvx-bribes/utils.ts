import BigNumber from 'bignumber.js';
import { formatWithoutExtraZeros, numberWithCommas, unscale } from '../../mobx/utils/helpers';

export function formatBalance(balance: BigNumber): string {
  return numberWithCommas(formatWithoutExtraZeros(unscale(balance, 18)));
}
