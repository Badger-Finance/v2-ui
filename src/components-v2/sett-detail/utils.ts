import dayjs from 'dayjs';
import utc from 'dayjs/plugin/utc';
import { ChartMode } from '../../mobx/model/setts/sett-charts';
import { DelaySeverity } from '../../mobx/model/setts/sett-rewards';
import { SettDetailMode } from '../../mobx/model/setts/sett-detail';

dayjs.extend(utc);

export const ChartModeTitles = {
	[ChartMode.value]: 'Sett Value',
	[ChartMode.ratio]: 'Token Ratio',
	[ChartMode.accountBalance]: 'Your Holdings',
};

export const SettModeTitles = {
	[SettDetailMode.settInformation]: 'Vault Info',
	[SettDetailMode.userInformation]: 'User Info',
};

export const calculateDelaySeverity = (delay: number): DelaySeverity => {
	if (delay >= 4) {
		return DelaySeverity.high;
	}

	if (delay >= 2) {
		return DelaySeverity.medium;
	}

	return DelaySeverity.none;
};

export const calculateDifferenceInHoursFromCycle = (cycle: Date): number => {
	return Math.abs(dayjs(cycle).diff(dayjs(), 'hours'));
};
