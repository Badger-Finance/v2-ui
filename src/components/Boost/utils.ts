import { makeStyles } from '@material-ui/core/styles';
import { ClassNameMap } from '@material-ui/core/styles/withStyles';
import React from 'react';

type ComparisonConfig = {
  toCompareValue: number;
  toBeComparedValue: number;
  greaterCaseColor: string;
  lessCaseColor: string;
  defaultColor: string;
};

export const getColorFromComparison = ({
  toCompareValue,
  toBeComparedValue,
  greaterCaseColor,
  lessCaseColor,
  defaultColor,
}: ComparisonConfig): string => {
  if (toCompareValue > toBeComparedValue) return greaterCaseColor;
  if (toCompareValue < toBeComparedValue) return lessCaseColor;

  return defaultColor;
};

export const useAssetInputStyles = (
  currentValue: string,
  toCompare = 0,
): ((props?: React.CSSProperties) => ClassNameMap<'assetColor'>) => {
  return makeStyles((theme) => {
    const defaultColor = currentValue ? theme.palette.text.primary : theme.palette.text.secondary;
    const fontColor = getColorFromComparison({
      toCompareValue: Number(currentValue),
      toBeComparedValue: toCompare,
      greaterCaseColor: '#74D189',
      lessCaseColor: theme.palette.error.main,
      defaultColor,
    });

    return {
      assetColor: {
        color: fontColor,
      },
    };
  });
};
