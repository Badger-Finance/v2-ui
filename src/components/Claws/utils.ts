import BigNumber from 'bignumber.js';
import { INVALID_REASON } from './index';

export interface ValidationParams {
	amount: string | BigNumber;
	maximum: BigNumber;
	minimum?: BigNumber;
}

export function validateAmountBoundaries({
	amount,
	maximum,
	minimum = new BigNumber('0'),
}: ValidationParams): INVALID_REASON | undefined {
	let error: INVALID_REASON | undefined = undefined;

	const input = new BigNumber(amount);
	const amountExceedsBalance = input.gt(maximum);
	const amountIsLessThanMinimum = input.lt(minimum);

	if (amountIsLessThanMinimum) error = INVALID_REASON.UNDER_MINIMUM;
	if (amountExceedsBalance) error = INVALID_REASON.OVER_MAXIMUM;

	console.log({
		input: input.toString(),
		minimum: minimum.toString(),
		maximum: maximum.toString(),
		amountExceedsBalance,
		amountIsLessThanMinimum,
		error,
	});

	return error;
}
