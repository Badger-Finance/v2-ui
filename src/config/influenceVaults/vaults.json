{
	"rewardFrequencies": [
		{
			"name": "bveCVX, BADGER",
			"value": "Each bi-weekly bribe sale"
		},
		{
			"name": "bcvxCRV",
			"value": "Each ~2hr cycle"
		}
	],
	"rewardFrequenciesModalInfo": {
		"title": "Reward Frequency",
		"body": [
			"bveCVX and BADGER rewards are distributed at the completion of each bi-weekly voting/bribing round, after bribes for that round have been sold. Convex locking rewards are distributed each ~2hr cycle as autocompounding ",
			"[bcvxCRV](convex-cvxcrv)",
			"."
		],
		"pointType": "list",
		"points": [
			["- 75% of bribe sales (after fees): sold for bveCVX"],
			["- 25% of bribe sales (after fees): sold for BADGER"],
			["- Convex vlCVX locking rewards: claimable as autocompounding bCvxCRV"],
			["- All rewards are claimable through the app"]
		]
	},
	"withdrawModalInfo": {
		"title": "CVX Withdrawable",
		"body": [
			"Each week, this vault locks batches of CVX tokens for a 16 week period. As vlCVX unlocks on a rolling basis, CVX becomes available to withdraw from the vault. Alternatively, bveCVX may be traded for CVX via the ",
			"[Curve pool](https://curve.fi/factory/52/)",
			" at any time."
		],
		"pointType": "list",
		"points": [
			["- CVX tokens are locked each Thursday just before 00:00 UTC"],
			["- Unlocked CVX is withdrawable from 00:01 UTC each Thursday until the next locking event"],
			[
				"- The unlocking schedule for bveCVX can be found on this ",
				"[Dune dashboard](https://dune.com/tianqi/Convex-Locked-CVX-V2(Sponsored-by-Badger))",
				"."
			]
		]
	},
	"perfomanceInfo": {
		"body": [
			[
				"This vault locks 100% of deposited Convex tokens for rolling periods of 16 weeks. Badger will use vlCVX to vote for bribes during each voting round, sell them, and emit the proceeds back to holders in the form of bveCVX (autocompounded), and claimable BADGER and bcvxCRV."
			],
			[
				"Unlike other Badger Vaults, bveCVX limits the times when users may withdraw their funds. Limited pre-unlock liquidity is available through this ",
				"[Curve Pool](https://curve.fi/factory/52/)",
				". Please carefully read the ",
				"[User Guide](https://docs.badger.com/badger-finance/vaults/vault-user-guides-ethereum/vote-locked-cvx)",
				" for more information. Details on the timing of CVX unlocks are available on this ",
				"[Dune dashboard](https://dune.com/tianqi/Convex-Locked-CVX-V2(Sponsored-by-Badger))",
				"."
			]
		],
		"liquity": [["% CVX Received from 10k "], ["[bveCVX swap](https://curve.fi/factory/52/)"]]
	},
	"feeInfo": {
		"fees": [
			{
				"bveCVX Liquidity Support": "5%",
				"BADGER Liquidity Support": "5%",
				"DAO Operations Fee": "5%"
			}
		],
		"feeModalInfo": {}
	}
}
