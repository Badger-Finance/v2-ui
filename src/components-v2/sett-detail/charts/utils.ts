import dayjs from 'dayjs';
import utc from 'dayjs/plugin/utc';
import { Sett } from '../../../mobx/model/setts/sett';
import { fetchSettChartInformation } from '../../../mobx/utils/apiV2';
import { SettSnapshotGranularity } from '../../../mobx/model/setts/sett-snapshot';
import { SettChartData } from '../../../mobx/model/setts/sett-charts';

dayjs.extend(utc);

export enum SettChartTimeframe {
	'day' = 1,
	'week' = 7,
	'month' = 30,
}

/**
 * Fetches chart information since the provided timeframe until the current date
 * @param sett
 * @param timeframe
 */
export const fetchSettChart = async (sett: Sett, timeframe: SettChartTimeframe): Promise<SettChartData[] | null> => {
	const timeframeDays = 1 * timeframe;
	const isDayTimeFrame = timeframe === SettChartTimeframe.day;

	// if timeframe is just one day then we want the granularity to be hours
	const granularity = isDayTimeFrame ? SettSnapshotGranularity.HOUR : SettSnapshotGranularity.DAY;

	const now = dayjs().utc(); // query until current date
	const from = dayjs(now).subtract(timeframeDays, 'days').utc();

	const fetchedData = await fetchSettChartInformation(sett.vaultToken, from.toDate(), now.toDate(), granularity);

	if (!fetchedData) {
		return null;
	}

	// data needs to be ascending sorted
	return fetchedData
		.sort((a, b) => a.timestamp - b.timestamp)
		.map((d) => ({ ...d, timestamp: new Date(d.timestamp) }));
};
