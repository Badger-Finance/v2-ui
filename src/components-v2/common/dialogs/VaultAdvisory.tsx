import { VaultDTO } from '@badger-dao/sdk';
import { AdvisoryType } from 'mobx/model/vaults/advisory-type';
import React from 'react';

import ChadgerVaultAdvisory from './ChadgerVaultAdvisory';
import LockingVaultAdvisory from './LockingVaultAdivsory';
import RemunerationVaultAdvisory from './RemunerationVaultAdvisory';

interface Props {
  vault: VaultDTO;
  type: AdvisoryType;
  accept: () => void;
}

const VaultAdvisory = ({ vault, type, accept }: Props): JSX.Element | null => {
  let advisory: JSX.Element | null;
  switch (type) {
    case AdvisoryType.ConvexLock:
      advisory = (
        <LockingVaultAdvisory
          accept={accept}
          vault={vault}
          lockingWeeks={16}
          learnMoreLink="https://docs.badger.com/badger-finance/sett-user-guides/blcvx-locked-convex"
        />
      );
      break;
    case AdvisoryType.VaultLock:
      advisory = <LockingVaultAdvisory vault={vault} accept={accept} lockingWeeks={16} />;
      break;
    case AdvisoryType.Remuneration:
      advisory = <RemunerationVaultAdvisory accept={accept} />;
      break;
    case AdvisoryType.Chadger:
      advisory = <ChadgerVaultAdvisory accept={accept} />;
      break;
    case AdvisoryType.None:
    default:
      advisory = null;
  }
  return advisory;
};

export default VaultAdvisory;
