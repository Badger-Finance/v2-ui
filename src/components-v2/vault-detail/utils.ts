import { Vault, VaultData } from '@badger-dao/sdk';
import dayjs from 'dayjs';
import utc from 'dayjs/plugin/utc';
import { ChartMode } from '../../mobx/model/vaults/vault-charts';
import { DelaySeverity } from '../../mobx/model/vaults/vault-rewards';

dayjs.extend(utc);

export const ChartModeTitles: Record<string, string> = {
	[ChartMode.Value]: 'Sett Value',
	[ChartMode.Ratio]: 'Token Ratio',
	[ChartMode.AccountBalance]: 'My Holdings',
	[ChartMode.BoostMultiplier]: 'Badger Boost',
};

export const calculateDelaySeverity = (delay: number): DelaySeverity => {
	if (delay >= 4) {
		return DelaySeverity.high;
	}

	if (delay >= 2) {
		return DelaySeverity.medium;
	}

	return DelaySeverity.none;
};

export const calculateDifferenceInHoursFromCycle = (cycle: Date): number => {
	return Math.abs(dayjs(cycle).diff(dayjs(), 'hours'));
};

export function defaultVaultBalance(vault: Vault): VaultData {
	return {
		address: vault.vaultToken,
		name: vault.name,
		symbol: vault.asset,
		pricePerFullShare: vault.pricePerFullShare,
		balance: 0,
		value: 0,
		earnedBalance: 0,
		earnedValue: 0,
		depositedBalance: 0,
		withdrawnBalance: 0,
		tokens: [],
		earnedTokens: [],
	};
}

export interface Balance {
	balance: number;
}

export function hasBalance(input?: Balance): boolean {
	if (!input) {
		return false;
	}
	return input.balance > 0;
}
