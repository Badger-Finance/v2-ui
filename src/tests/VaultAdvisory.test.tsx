import React from 'react';
import '@testing-library/jest-dom';
import { checkSnapshot } from './utils/snapshots';
import { AdvisoryType } from 'mobx/model/vaults/advisory-type';
import VaultAdvisory from 'components-v2/common/dialogs/VaultAdvisory';

describe('VaultAdvisory', () => {
	describe('Shows the proper vault advisory if one exists, or no advisory', () => {
		it.each([
			[AdvisoryType.ConvexLock],
			[AdvisoryType.Remuneration],
			[AdvisoryType.Chadger],
			[AdvisoryType.None],
			['Invalid'], // purposeful bad enum case
		])('Displays the expected %s vault advisory', (advisoryType: string) => {
			checkSnapshot(<VaultAdvisory accept={() => {}} type={advisoryType as AdvisoryType} />);
		});
	});
});
