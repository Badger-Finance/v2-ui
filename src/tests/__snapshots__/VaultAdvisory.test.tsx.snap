// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VaultAdvisory Shows the proper vault advisory if one exists, or no advisory Displays the expected Chadger vault advisory 1`] = `
<div>
  <div
    class="makeStyles-vaultAdvisoryContainer-1"
  >
    <div
      class="makeStyles-childContainer-2"
    >
      <p>
        This is a Chadger Vault, meaning a highly experimental vault that builds on new protocols. This vault carries higher levels of liquidity and smart contract risk than normal Badger Vaults.
      </p>
      <p>
        Ape at your own risk.
      </p>
    </div>
    <label
      class="MuiFormControlLabel-root-3"
    >
      <span
        aria-disabled="false"
        class="MuiButtonBase-root-28 MuiIconButton-root-19 PrivateSwitchBase-root-15 MuiCheckbox-root-9 MuiCheckbox-colorPrimary-13 MuiIconButton-colorPrimary-23"
      >
        <span
          class="MuiIconButton-label-27"
        >
          <input
            class="PrivateSwitchBase-input-18"
            data-indeterminate="false"
            name="read"
            type="checkbox"
            value=""
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root-31 MuiSvgIcon-fontSizeSmall-38"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root-101"
        />
      </span>
      <span
        class="MuiTypography-root-40 MuiFormControlLabel-label-8 MuiTypography-body1-42"
      >
        I have read and understand the above vault advisory
      </span>
    </label>
    <button
      aria-label="Deposit"
      class="MuiButtonBase-root-28 MuiButton-root-72 MuiButton-contained-80 WithStyles(ForwardRef(Button))-root-70 WithStyles(ForwardRef(Button))-root-71 MuiButton-containedSecondary-82 MuiButton-containedSizeLarge-92 MuiButton-sizeLarge-94 MuiButton-disabled-85 MuiButton-fullWidth-95 MuiButtonBase-disabled-29"
      disabled=""
      tabindex="-1"
      type="button"
    >
      <span
        class="MuiButton-label-73"
      >
        Accept
      </span>
    </button>
  </div>
</div>
`;

exports[`VaultAdvisory Shows the proper vault advisory if one exists, or no advisory Displays the expected ConvexLock vault advisory 1`] = `
<div>
  <div
    class="makeStyles-vaultAdvisoryContainer-1"
  >
    <div
      class="makeStyles-childContainer-2"
    >
      <p>
        This vault locks Convex in a staking contract that is 16 weeks long.
      </p>
      <p>
        The Total Withdrawable Amount in the vault may be lower than your balance depending on what is currently available and will fluctuate with deposits and withdraws from the vault as well as locks and unlocks from the contract.
      </p>
      <p>
        Any vault withdraws will have an opportunity to utilize the freshly deposited liquidity before it gets locked on a first-come-first-serve basis.
      </p>
      <div
        class="makeStyles-linkContainer-3"
      >
        <a
          class="MuiTypography-root-10 MuiLink-root-4 MuiLink-underlineHover-6 MuiTypography-colorPrimary-33"
          href="https://docs.badger.com/badger-finance/sett-user-guides/blcvx-locked-convex"
          rel="noreferrer"
          target="_blank"
        >
          Learn More
        </a>
      </div>
    </div>
    <label
      class="MuiFormControlLabel-root-40"
    >
      <span
        aria-disabled="false"
        class="MuiButtonBase-root-65 MuiIconButton-root-56 PrivateSwitchBase-root-52 MuiCheckbox-root-46 MuiCheckbox-colorPrimary-50 MuiIconButton-colorPrimary-60"
      >
        <span
          class="MuiIconButton-label-64"
        >
          <input
            class="PrivateSwitchBase-input-55"
            data-indeterminate="false"
            name="read"
            type="checkbox"
            value=""
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root-68 MuiSvgIcon-fontSizeSmall-75"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root-108"
        />
      </span>
      <span
        class="MuiTypography-root-10 MuiFormControlLabel-label-45 MuiTypography-body1-12"
      >
        I have read and understand the above vault advisory
      </span>
    </label>
    <button
      aria-label="Deposit"
      class="MuiButtonBase-root-65 MuiButton-root-79 MuiButton-contained-87 WithStyles(ForwardRef(Button))-root-77 WithStyles(ForwardRef(Button))-root-78 MuiButton-containedSecondary-89 MuiButton-containedSizeLarge-99 MuiButton-sizeLarge-101 MuiButton-disabled-92 MuiButton-fullWidth-102 MuiButtonBase-disabled-66"
      disabled=""
      tabindex="-1"
      type="button"
    >
      <span
        class="MuiButton-label-80"
      >
        Accept
      </span>
    </button>
  </div>
</div>
`;

exports[`VaultAdvisory Shows the proper vault advisory if one exists, or no advisory Displays the expected Invalid vault advisory 1`] = `<div />`;

exports[`VaultAdvisory Shows the proper vault advisory if one exists, or no advisory Displays the expected None vault advisory 1`] = `<div />`;

exports[`VaultAdvisory Shows the proper vault advisory if one exists, or no advisory Displays the expected Remuneration vault advisory 1`] = `
<div>
  <div
    class="makeStyles-vaultAdvisoryContainer-1"
  >
    <div
      class="makeStyles-childContainer-2"
    >
      <p>
        Withdrawing from your remBADGER position is a one-way transaction.
      </p>
      <p>
        Once withdrawn, you will not be able to re-deposit, and as a result, agree to forfeit any future accrued Badger rewards emitted throughout the designated restitution period as outlined in
         
        <a
          class="MuiTypography-root-9 MuiLink-root-3 MuiLink-underlineHover-5 MuiTypography-colorPrimary-32"
          href="https://forum.badger.finance/t/bip-80-restitution-of-non-recoverable-assets-via-rembadger-sett/5362"
        >
          BIP 80
        </a>
        .
      </p>
      <p>
        The amount of redeemable remBADGER will depend on when itâ€™s withdrawn and will increase over time as additional funds are added to the vault by BadgerDAO.
      </p>
      <div
        class="makeStyles-linkContainer-39"
      >
        <a
          class="MuiTypography-root-9 MuiLink-root-3 MuiLink-underlineHover-5 MuiTypography-colorPrimary-32"
          href="https://docs.badger.com/badger-finance/sett-user-guides/blcvx-locked-convex"
          rel="noreferrer"
          target="_blank"
        >
          Learn More
        </a>
      </div>
    </div>
    <label
      class="MuiFormControlLabel-root-40"
    >
      <span
        aria-disabled="false"
        class="MuiButtonBase-root-65 MuiIconButton-root-56 PrivateSwitchBase-root-52 MuiCheckbox-root-46 MuiCheckbox-colorPrimary-50 MuiIconButton-colorPrimary-60"
      >
        <span
          class="MuiIconButton-label-64"
        >
          <input
            class="PrivateSwitchBase-input-55"
            data-indeterminate="false"
            name="read"
            type="checkbox"
            value=""
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root-68 MuiSvgIcon-fontSizeSmall-75"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root-108"
        />
      </span>
      <span
        class="MuiTypography-root-9 MuiFormControlLabel-label-45 MuiTypography-body1-11"
      >
        I have read and understand the above vault advisory
      </span>
    </label>
    <button
      aria-label="Deposit"
      class="MuiButtonBase-root-65 MuiButton-root-79 MuiButton-contained-87 WithStyles(ForwardRef(Button))-root-77 WithStyles(ForwardRef(Button))-root-78 MuiButton-containedSecondary-89 MuiButton-containedSizeLarge-99 MuiButton-sizeLarge-101 MuiButton-disabled-92 MuiButton-fullWidth-102 MuiButtonBase-disabled-66"
      disabled=""
      tabindex="-1"
      type="button"
    >
      <span
        class="MuiButton-label-80"
      >
        Accept
      </span>
    </button>
  </div>
</div>
`;
